src/analytics/api/utils/data_formatters.py

```diff
@@ -280,3 +288,26 @@ def device_category_to_str(device_category: str) -> str:
         pass
 
     return ""
+
+
+def filter_non_private_entities(entities: list, entity_type: Entity) -> list:
+    source = "cohorts" if entity_type == Entity.DEVICES else "grids"
+
+    try:
+        response = requests.post(
+            url=f"{Config.AIRQO_API_BASE_URL}/devices/{source}/filterNonPrivate{entity_type.value.capitalize()}",
+            json={entity_type.value: entities},
+            params={"token": Config.AIRQO_API_TOKEN}
+        )
+        response.raise_for_status()
+
+        data = response.json()
+        if data.get("success"):
+            return data.get(entity_type.value, [])
+    except requests.exceptions.HTTPError as err:
+        print(f"HTTP Error: {err}")
+    except requests.exceptions.RequestException as e:
+        print(f"Error: {e}")
+
+    # TODO: Remove once @Martin updates endpoint to support other ID format
+    return entities
```

Comment:
@Baalmart , for now I've set it to return the original devices (no flitering) as you work on the enhancement to use the other version of the `device_IDs` as we had discussed on slack

Reply:
thanks @Mnoble-19